data-prepper-core/src/main/java/org/opensearch/dataprepper/core/sourcecoordination/LeaseBasedSourceCoordinator.java

```diff
@@ -351,6 +351,30 @@ public void giveUpPartition(final String partitionKey, final Instant priorityTim
         giveUpPartitionInternal(partitionKey, priorityTimestamp);
     }
 
+    @Override
+    public void giveUpPartition(final String partitionKey, final Instant priorityTimestamp,final Integer maxRetries) {
+        final long backoffMillis = 200;
+        for (int attempt = 1; attempt <= maxRetries; attempt++) {
+            try {
+                giveUpPartitionInternal(partitionKey, priorityTimestamp);
+                return;
+            } catch (PartitionNotOwnedException e) {
+                if (attempt == maxRetries) {
+                    throw e;
+                }
+                LOG.warn("Partition {} not owned on attempt {}/{}. Retrying...", partitionKey, attempt, maxRetries);
```

Comment:
Let's avoid ellipses. When will it retry? Is it trying now? 

Maybe we can say something like: `Will retry giving up the partition`.